### PR TITLE
Improvement Router Method

### PR DESCRIPTION
**Pull request by developer-dlio**
Enable multiple get and post methods without having to specify path again.

### CODE REVIEW

**Comment by developer-dlio** [comment]
Sample!

image

File: src/Horse.Core.pas [hunk 1]
```diff
@@ -5,6 +5,80 @@ interface
 uses System.SysUtils, Web.HTTPApp, Horse.Router;
 
 type
+  THorseCore = class;
+
+  IHorseCoreRoute = interface
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:10]: Create an unit for IHorseCoreRoute: Horse.Core.Route.Intf.pas

File: src/Horse.Core.pas [hunk 2]
```diff
@@ -5,6 +5,80 @@ interface
...
+    function Post(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Post(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Delete(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Delete(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Delete(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Delete(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+  end;
+
+  THorseCoreRoute = class(TInterfacedObject, IHorseCoreRoute)
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:43]: Create an unit for THorseCoreRoute: Horse.Core.Route.pas

File: src/Horse.Core.pas [hunk 3]
```diff
@@ -5,6 +5,80 @@ interface
...
+  private
+    FPath:string;
+    FHorseCore:THorseCore;
+
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:47]: Remove empty line

File: src/Horse.Core.pas [hunk 4]
```diff
@@ -5,6 +5,80 @@ interface
...
+    function Patch(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Patch(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Patch(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Head(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Post(ACallback: THorseCallback):IHorseCoreRoute; overload;
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:71]: In all functions, place a space after the colon of the function:

`function Post(ACallback: THorseCallback):IHorseCoreRoute; overload;`

to

`function Post(ACallback: THorseCallback): IHorseCoreRoute; overload;`

File: src/Horse.Core.pas [hunk 5]
```diff
@@ -16,7 +90,9 @@   THorseCore = class
     destructor Destroy; override;
     constructor Create; overload;
     class destructor UnInitialize;
+
     property Routes: THorseRouterTree read FRoutes write FRoutes;
+    function Route(APath:string):IHorseCoreRoute;
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:95]: In all parameters, place a space after the colon:

`function Route(APath:string):IHorseCoreRoute;` 

to

`function Route(APath: string): IHorseCoreRoute;`

File: src/Horse.Core.pas [hunk 6]
```diff
@@ -170,6 +246,11 @@ procedure THorseCore.RegisterRoute(AHTTPType: TMethodType; APath: string; ACallb
   FRoutes.RegisterRoute(AHTTPType, APath, ACallback);
 end;
 
+function THorseCore.Route(APath: string): IHorseCoreRoute;
+begin
+  Result:= THorseCoreRoute.Create(APath, Self);
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:251]: Add space before and after :=

File: src/Horse.Core.pas [hunk 7]
```diff
@@ -286,4 +367,156 @@ procedure THorseCore.Put(APath: string; ACallbacks: array of THorseCallback; ACa
   Put(APath, ACallback);
 end;
 
+{ THorseCoreRoute }
+
+constructor THorseCoreRoute.Create(APath:string; AHorseCore: THorseCore);
+begin
+  FPath      := APath;
```
Comment (viniciussanchez) [inline src/Horse.Core.pas:374]: Add **one** space before and after :=

File: src/Horse.Core.Route.pas [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+interface
+
+uses
+  Horse.Core.Route.Intf, Horse.Router, Horse.Core;
+
+type
+  THorseCoreRoute = class(TInterfacedObject, IHorseCoreRoute)
+  private
+    FPath:string;
```
Comment (viniciussanchez) [inline src/Horse.Core.Route.pas:11]: Change to:
`FPath: string;`

File: src/Horse.Core.Route.pas [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+    FHorseCore:THorseCore;
```
Comment (viniciussanchez) [inline src/Horse.Core.Route.pas:12]: Change to:
`FHorseCore: THorseCore;`

File: src/Horse.Core.Route.Intf.pas [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+interface
+
+uses
+  Horse.Router;
+
+type
+  IHorseCoreRoute = interface
+    ['{39ABE601-E812-40B4-8857-F14C938B9773}']
+    function Get(ACallback: THorseCallback):IHorseCoreRoute; overload;
```
Comment (viniciussanchez) [inline src/Horse.Core.Route.Intf.pas:11]: Change to in all functions:

`function Get(ACallback: THorseCallback): IHorseCoreRoute; overload;`

**Pull request merged**